src/google/protobuf/generated_message_tctable_impl.h

```diff
@@ -95,11 +96,11 @@ namespace internal {
 
 #ifndef NDEBUG
 template <size_t align>
-#ifndef _MSC_VER
-[[noreturn]]
-#endif
 void AlignFail(uintptr_t address) {
   GOOGLE_LOG(FATAL) << "Unaligned (" << align << ") access at " << address;
+
+  // Explicit abort to let compilers know this function does not return
```

Comment:
Replace this with `// This abort() is unreachable; it is only here to silence compiler warnings that do not know that `FATAL` always terminates the program.`.

Reply:
Done.